superset-frontend/src/explore/components/controls/VizTypeControl/VizTypeGallery.tsx

```diff
@@ -545,7 +589,18 @@ export default function VizTypeGallery(props: VizTypeGalleryProps) {
     if (searchInputValue.trim() === '') {
       return [];
     }
-    return fuse.search(searchInputValue).map(result => result.item);
+    return fuse
+      .search(searchInputValue)
+      .map(result => result.item)
+      .sort((a, b) => {
```

Comment:
This `sort` is new. Before we were using the numeric weighting. Does the weighting still work, or does this search "win" in terms of ranking things?

Reply:
Yes, now we use `CHART_LABEL_ORDER` array to determine the search weight, and the larger the `index`, the larger the `weight`

Reply:
Maybe instead of using this sort based on the label's place in the array order, we can define the labels with some basic (flexible/configurable) weighting, like this
```js
const chartLabels = {
    Featured: {
        weight: 0.3
    },
    Depracated: {
        weight: -0.3
    }
}
```

Reply:
This is better (and I think is sufficient for the feature), but I wish there was a way to avoid using `sort` and instead have this be just one more _factor_ in the search result, like the [weight for keys](https://fusejs.io/examples.html#weighted-search) that `fuse` supports. This might be a bit heavy-handed in terms of search weighting, but... the feature works!